### PR TITLE
Exposes a function to check if a string is a URL.

### LINKED ISSUE #72: Expose URL Parsing to Library Users
Hello!

As part of a user-safety project, I'm integrating google/safebrowsing into my employer's chat product as a means of filtering out harmful links. One difficulty we've run into is that our link parsing doesn't exactly line up with the library's. Specifically, we are using `https://github.com/mvdan/xurls` to find links in a chat message, and we then pass all of those links into the `SafeBrowser.LookupURLS` method. This has resulted in some infrequent, but regular edge-cases where the link parsing regex provided by `xurls`has a more expansive definition of a link than this library, leading to a `safebrowser : invalid path` errors.

I think the best, and easiest remediation for this would be if this library exposed a simple `IsURL` function, or provided an `ExtractUrls` method which would pull out all of the URLs in a string. That way, we could either filter out those URL-like strings which would cause an error, or just use this library's idea of what a URL is throughout our codebase.

Regards,
Ore Babarinsa.

### PR DESCRIPTION
**Pull request by colonelxc**
This makes it possible to check if a string will pass URL parsing before
calling LookupURLs. This can help identify if one string in a batch will
cause an error that will abort the LookupURLs call.

A pitfall of this method is that any series of words without punctuation
is considered a valid URL, as the code will turn spaces into '%20's and
consider the whole string as a long hostname.

Fixes #72

### CODE REVIEW

File: urls_test.go [hunk 1]
```diff
@@ -21,6 +21,78 @@ import (
 	"testing"
 )
 
+func TestIsURL(t *testing.T) {
```
Comment (dsnet) [inline urls_test.go]: I don't really see the value of `TestIsURL`. As it currently stands, it is just change detector for the unexported parseURL function, which is itself unit tested.

I'm not opposed to keeping it though.

**Comment by dsnet** [comment]
Alternatively, you could call it `ValidURL`.

File: urls_test.go (hunk 1 above)
Comment (colonelxc) [reply urls_test.go]: Agreed, all the same test cases as before.

**Comment by dsnet** [comment]
Still LGTM.

**Pull request merged**